[+] Chuni Userbox with Assets

raymonable opened the pull request:
it's finally done

a few notes:
 - chinese translations need to be filled in
 - there are known issues in safari
 - no profile userbox support until the API to access the items of other people's userboxes is added
 - trophy colors are incorrect because it's unavailable in the API, scraping game files for it could be feasible but isn't implemented right now
 - fallback for avatar clothing being missing in files is the default wear, may confuse some players?

## Summary by Sourcery

实现带有铭牌和头像自定义的Chuni用户框显示。

新功能：
- 显示可自定义铭牌和头像的Chuni用户框。

测试：
- 为新的用户框功能添加UI元素。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Implement Chuni userbox display with nameplate and avatar customization.

New Features:
- Display Chuni userboxes with customizable nameplates and avatars.

Tests:
- Add UI elements for new userbox features.

</details>

raymonable committed:
[+] Chuni Userbox with Assets

Co-authored-by: split / May  <split@split.pet>

raymonable commented:
![image](https://github.com/user-attachments/assets/67481b76-b4d9-4613-a910-846acebf6b62)

nbitzz commented:
Hi, question: are the postinstall scripts from `@parcel/watcher` supposed to be left untrusted, or am I just doing something wrong? I don't regularly use Bun, so need advice here.

nbitzz committed:
refactor: :recycle: replace "CHUNITHM" string with "AquaDX"

recommendation from @raymonable

nbitzz committed:
style: linebreaks

nbitzz committed:
docs: :memo: add TSDoc comments to functions in the DDS class

clansty commented:
It seems that the "input url" feature is removed in i18n files. Is this feature removed?

I think we can keep both "input url" and "select game folder" and let the user to choose

nbitzz commented:
> It seems that the "input url" feature is removed in i18n files. Is this feature removed?
> 
> I think we can keep both "input url" and "select game folder" and let the user to choose

It looks like it was removed. Raymond's asleep, but since he seems to want to get this merged so quickly, I'll do my best to look at what was there previously and reimplement it myself...

nbitzz commented:
Should the `HAS_USERBOX_ASSETS` `{#if}` block encapsulate the new AquaBox options aswell?

**Edit:** I've had it encapsulate the AquaBox options as well; let me know if you wish for this change to be reverted

nbitzz committed:
revert: :rewind: Reverse decision to remove the classic UserBox preview

Adds back the classic UserBox preview when AquaBox is disabled / unavailable

raymonable commented:
> It seems that the "input url" feature is removed in i18n files. Is this feature removed?

i figured not many people got use out of it being there, since there's no documentation on how to setup the files-- it isn't very useful for the average public server player

clansty commented:
> i figured not many people got use out of it being there, since there's no documentation on how to setup the files-- it isn't very useful for the average public server player

I think they may only need an url published by some maybe official one in discord

raymonable committed:
style: make nameplates fit better

raymonable committed:
refactor: move DDS cache

moved the DDS cache from dds.ts to ddsCache.ts and added caching for scaled images

Co-authored-by: split / May <split@split.pet>

hykilpikonna commented:
> Hi, question: are the postinstall scripts from `@parcel/watcher` supposed to be left untrusted, or am I just doing something wrong? I don't regularly use Bun, so need advice here.

Uhh I don't think it matters, like I did not run the postinstall and it worked fine.

hykilpikonna commented on AquaNet/src/libs/i18n/en_ref.ts:
```diff
@@ -179,6 +179,17 @@ export const EN_REF_USERBOX = {
+  'userbox.new.activate_update': 'Update AquaBox (game files required)',
+  'userbox.new.activate': 'Use AquaBox',
+  'userbox.new.activate_desc': 'Enable displaying UserBoxes with their nameplate & avatar',
+  'userbox.new.error.invalidFolder': 'The folder you selected is invalid. Ensure that your game\'s version is Lumi or newer and that the "A001" option pack is present.'
```
Do you mean A000?

raymonable replied:
yes, my mistake

hykilpikonna commented on AquaNet/src/libs/userbox/userbox.ts:
```diff
@@ -0,0 +1,180 @@
+export async function userboxFileProcess(folder: FileSystemEntry, progressUpdate: (progress: number, progressString: string) => void): Promise<string | null> {
+    if (!isDirectory(folder))
+        return t("userbox.new.error.invalidFolder")
+    if (!(await validateDirectories(folder, "bin/option")) || !(await validateDirectories(folder, "data/A000")))
```
I think the logic here might be incorrect. Currently it requires both of these paths to be present, but they should be optional if only one is present?

raymonable replied:
i mean data/A000 is required 100% but bin/option might not be? but there's such an unlikely chance that someone using game data won't have an option
